docs/GettingStarted/Tutorials/Introduction/Examples.md

```diff
@@ -0,0 +1,145 @@
+      evenIdx = (int)i;
+      break;
+    }
+  }
+
+  for (unsigned i = 1; i < N; i += 2) {
+    if (a[i] == search) {
+      oddIdx = (int)i;
+      break;
+    }
+  }
+
+  int done = -1;
+  if (evenIdx != -1)
+    done = evenIdx;
+  else if (oddIdx != -1)
+    done = oddIdx;
+
+  return done;
+}
+
+int main(void) {
+  in_int_t search = 55;
+  in_int_t a[N];
+  for (int i = 0; i < N; i++)
+    a[i] = i;
+  CALL_KERNEL(binary_search, search, a);
+  return 0;
+}
+```
+The conditional statements in the for loops make control flow unpredictable. Conventional HLS tools would avoid all risks by assuming a serial dependency between iterations of the loop. Dynamatic takes the risks while producing funcitonally correct and efficient dataflow circuits.
+#### Launching Dynamatic
+If you haven't added Dynamatic to path, navigate to the directory where you cloned Dynamatic and run the command below:
+```sh
+./bin/dynamatic
+```
+The Dynamatic frontend would be displayed as follows
+```sh
+username:~/Dynamatic/dynamatic$ ./bin/dynamatic
+================================================================================
+============== Dynamatic | Dynamic High-Level Synthesis Compiler ===============
+======================== EPFL-LAP - v2.0.0 | March 2024 ========================
+================================================================================
+
+
+dynamatic> 
+```
+#### Set the Path to the C Target C File
+Use the `set-src` command to direct Dynamatic to the file you want to synthesize into RTL
+```sh
+dynamatic> set-src integration-test/binary_search/binary_search.c
+```
+#### Compile the C File to a Lower Intermediate Representation
+You can choose the **buffer placement algorithm** with the `--buffer-algorithm` flag. For this example, we use `fpga20`, a throughput driven algorithm. 
+> [!TIP]  
+> If you are not sure which options are available for the compile command, add anything after it and hit enter to see the options e.g compile --
+```sh
+dynamatic> compile --buffer-algorithm fpga20
+[INFO] Compiled source to affine
+[INFO] Ran memory analysis
+[INFO] Compiled affine to scf
+[INFO] Compiled scf to cf
+[INFO] Applied standard transformations to cf
+[INFO] Applied Dynamatic transformations to cf
+[INFO] Compiled cf to handshake
+[INFO] Applied transformations to handshake
+[INFO] Built kernel for profiling
+[INFO] Ran kernel for profiling
+[INFO] Profiled cf-level
+[INFO] Running smart buffer placement with CP = 4.000 and algorithm = 'fpga20'
+[INFO] Placed smart buffers
+[INFO] Canonicalized handshake
+[INFO] Created binary_search DOT
+[INFO] Converted binary_search DOT to PNG
+[INFO] Created binary_search_CFG DOT
+[INFO] Converted binary_search_CFG DOT to PNG
+[INFO] Lowered to HW
+[INFO] Compilation succeeded
+```
+> [!TIP]  
+> Two PNG files are generated at compile time, `kernel_name.png` and `kernel_name_CFG.png`, allowing you to have a preview of your circuit and its control flow graph generated by Dynamatic.
```

Comment:
This is the tutorial doc so showing the real picture here is very helpful I think